### ISSUE TITLE
building of an individual test fails with sdk\5.0

### ISSUE THREAD

**Issue by sandreenko** [dotnet/coreclr#27039]
If you install Core-SDK 5.0 from https://github.com/dotnet/core-sdk#installers-and-binaries (you need it if you want to run perfomance benchmarks from with current CoreCLR master)
and then try to build an individual CoreCLR test with:
`C:\git\coreclr\tests\src\JIT\Regression\JitBlue\DevDiv_150586>C:\git\coreclr\.dotnet\dotnet.exe msbuild /p:__BuildOS=Windows_NT /p:__BuildArch=x64 /p:__BuildType=Checked  DevDiv_150586.ilproj`

you will see:
```
Microsoft (R) Build Engine version 16.3.0-preview-19456-02+ee8294b55 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
C:\Program Files\dotnet\sdk\5.0.100-alpha1-014826\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(262,5): error NETSDK1073: The FrameworkReference 'Microsoft.NETCore.App' was not recognized [C:\git\coreclr\tests\src\JIT\Regression\JitBlue\DevDiv_150586\DevDiv_150586.ilproj]
```

A workaround that I found for it is to add `<GenerateErrorForMissingTargetingPacks>false</GenerateErrorForMissingTargetingPacks>` to the project file (into a <PropertyGroup>), then it will build fine:

```
C:\git\coreclr\tests\src\JIT\Regression\JitBlue\DevDiv_150586>C:\git\coreclr\.dotnet\dotnet.exe msbuild /p:__BuildOS=Windows_NT /p:__BuildArch=x64 /p:__BuildType=Checked  DevDiv_150586.ilproj
Microsoft (R) Build Engine version 16.3.0-preview-19456-02+ee8294b55 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
  DevDiv_150586 -> C:\git\coreclr\bin\tests\Windows_NT.x64.Checked\JIT\Regression\JitBlue\DevDiv_150586\DevDiv_150586\DevDiv_150586.dll
```

The full test build is not affected (via `build.cmd x64 checked`), if you delete 5.0 (and stay on 3.0/3.1) then it works fine again.

**Comment by sandreenko** [comment]
I have checked that it reproduces on a clean VM, @echesakov you have changed the tests recently, could you take a look?

cc @dotnet/jit-contrib

**Comment by sandreenko** [comment]
PTAL @sbomer

**Comment by sbomer** [comment]
I believe this is caused by https://github.com/dotnet/arcade/issues/3763, which will be fixed with https://github.com/dotnet/arcade/pull/3975.

**Comment by sandreenko** [comment]
Fixed by #26784.
